src/protocol_edge.c

```diff
@@ -314,7 +356,19 @@ bool del_edge_h(meshlink_handle_t *mesh, connection_t *c, const char *request) {
 
 	/* Tell the rest about the deleted edge */
 
-	forward_request(mesh, c, request);
+
+	if(e->from->submesh && e->to->submesh && (e->from->submesh == e->to->submesh)) {
+		if(e->from->submesh) {
+			s = e->from->submesh;
+		} else {
+			s = e->to->submesh;
+		}
+
+		/* Tell the rest about the deleted edge */
+
+		forward_request(mesh, c, s, request);
```

Comment:
This should not be inside the `if`-statement: we also have to forward edges of nodes that are in the core mesh.

Reply:
The if condition is modified to also check if any of the node belongs to core mesh and if any of the node is core mesh node  it forwards the request. I have done this only to maintain the order of execution i.e only to delete that edge after forwarding it to other nodes. Other wise i will not have the edge handle to do the forwarding after deleting the edge from the graph i.e after calling edge_del() function